python/dgl/graphbolt/itemset.py

```diff
@@ -362,33 +362,19 @@ def __getitem__(self, index: Union[int, slice, Iterable[int]]):
                     break
             return data
         elif isinstance(index, Iterable):
-            # TODO[Mingbang]: Might have performance issue. Tests needed.
-            data = {key: [] for key in self._keys}
-            for idx in index:
-                if idx < 0:
-                    idx += self._length
-                if idx < 0 or idx >= self._length:
-                    raise IndexError(
-                        f"{type(self).__name__} index out of range."
-                    )
-                offset_idx = torch.searchsorted(self._offsets, idx, right=True)
-                offset_idx -= 1
-                idx -= self._offsets[offset_idx]
-                key = self._keys[offset_idx]
-                data[key].append(int(idx))
-            for key in self._keys:
-                indices = data[key]
-                if len(indices) == 0:
-                    del data[key]
+            index = torch.tensor(index).squeeze()
+            assert torch.all((index >= 0) & (index < self._length))
+            key_indices = (
+                torch.searchsorted(self._offsets, index, right=True) - 1
+            )
```

Comment:
```suggestion
            index = torch.tensor(index).squeeze().sort()[0]
            assert index[0] >= 0 and index[-1] < self._length
            offsets = (
                torch.searchsorted(index, self._offsets, right=False)
            )
```

Reply:
Is it necessary to sort the index? What if we want to reserve the order?

Reply:
We can then use the second output of sort to permute it back.